doc/function.md

```diff
@@ -1,5 +1,282 @@
 # Function
 
-You are reading a draft of the next documentation and it's in continuous update so
-if you don't find what you need please refer to:
-[C++ wrapper classes for the ABI-stable C APIs for Node.js](https://nodejs.github.io/node-addon-api/)
+The **Function** class provides a set of methods to create a function object in
+native code that can later be called from JavaScript. The created function is not
+automatically visible from JavaScript, instead it needs to be part of the add-on's
+module exports or be returned by one of the modules exported functions.
+
+In addition the `Function` class also provides methods that can be used to call
+functions that were created in JavaScript and passed to the native.
+
+The `Function` class inherits its behavior from the `Object` class (for more info
+see: [Object](object.md)).
+
+## Example
+
+```cpp
+#include <napi.h>
+
+using namespace Napi;
+
+Value Fn(const CallbackInfo& info) {
+  Env env = info.Env();
+  // ...
+  return String::New(env, "Hello World");
+}
+
+Object Init(Env env, Object exports) {
+  exports.Set(String::New(env, "fn"), Function::New(env, Fn));
+}
+
+NODE_API_MODULE(NODE_GYP_MODULE_NAME, Init)
+```
+
+The above code can be used from JavaScript as follows:
+
+```js
+const addon = require('./addon');
+addon.fn();
+```
+
+The `Function` class allows to call a JavaScript function object from a native
```

Comment:
Maybe we should rephrase this as "With the `Function` class it is possible to call..."